src/ethereum/api/Bridge.ts

```diff
@@ -31,51 +55,49 @@ export class Bridge {
   ) {
     const items: ItemAttributes[] = []
 
-    // Get collections from DB
-    const collectionIds = dbItems.reduce<string[]>((ids, dbItem) => {
-      if (dbItem.collection_id) {
-        ids.push(dbItem.collection_id)
-      }
-      return ids
-    }, [])
-    const dbResults = await Collection.findByIds(collectionIds)
-    const dbCollections = this.indexById(dbResults)
+    // To avoid multiple queries to the db, we will fetch all the items that match the blockchain_id and their collections
+    // to filter them later
+    let remoteDBItems = await Item.findByBlockchainIdsAndContractAddresses(
+      remoteItems.map((remoteItem) => ({
+        blockchainId: remoteItem.blockchainId,
+        collectionAddress: remoteItem.collection.id,
+      }))
+    )
 
-    // Get remote collections
-    const addresses = Object.values(dbCollections).map(
-      (dbCollection) => dbCollection.contract_address
+    // Get db collections from DB items
+    const collectionIds = [...dbItems, ...remoteDBItems].reduce<string[]>(
+      (ids, dbItem) => {
+        if (dbItem.collection_id) {
+          ids.push(dbItem.collection_id)
+        }
+        return ids
+      },
+      []
```

Comment:
I know it looks like I hate `.reduce` but...I do (?)

In all honesty, I think the method is not meant for these kinds of things and it just makes code way harder to read for no reason. For example if you want to go the functional route, you can:

```ts
[...dbItems, ...remoteDBItems].filter(item => !!item.collection_id).map(item => item.collection_id)
```

or just

```ts
const collectionIds = []
for(const item of [...dbItems, ...remoteDBItems]) {
  if(item.collection_id) {
    collectionIds.push(item.collection_id)
  }
}
```

In both these and another review where I suggested the same, you can see that doing a for...of usually is less code and way more readable

⚠️ This is my take on this, I have ways to justify it, but I'm open to chat about it